Web: Make s3 fields optional

kimlisa opened the pull request:
requires 
- [x] https://github.com/gravitational/teleport/pull/39912
- [x] e2e testing

Description:
- When creating AND editing integration: allow s3 fields to be optional, if user leaves it empty, render a warning
- When listing integrations without s3 fields defined, renders a tooltip that provides information to the user on how to update their thumbprint should they run into issues

listing:
<img width="1424" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/64669fed-30ba-46f0-838f-15525494b9c1">

listing, after clicking on button:
<img width="400" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/723ed5c3-42b7-40d3-a1b9-22815ffc369c">


editing:
<img width="753" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/eab7c215-55e0-4630-830e-c79a9c5b12de">

creating:
<img width="1051" alt="image" src="https://github.com/gravitational/teleport/assets/43280172/7ec53028-818e-413c-9247-2ed8825c65d9">


changelog: Amazon S3 fields are now optional when creating or editing AWS OIDC integration on the web UI

kimlisa committed:
Add outline-warn to alert banners

kimlisa committed:
Conditionally add s3 fields when generating script

kimlisa committed:
Allow s3 fields rules to ignore empty input conditionally

kimlisa committed:
Add hover tooltip for aws rows without s3 fields

kimlisa committed:
Show warning if user does not provide s3 fields when creating integration

kimlisa committed:
Show warning if user leaves s3 bucket fields empty when editing integration

kimlisa committed:
Don't prefill if input is empty

avatus reviewed (approved):
I didn't get to run it e2e yet but code looks good

kimlisa commented on web/packages/design/src/Alert/Alert.jsx:
```diff
@@ -56,8 +56,16 @@ const kind = props => {
-        background: fade(theme.colors.link, 0.1),
-        border: `${theme.radii[1]}px solid ${theme.colors.link}`,
+        background: fade(theme.colors.accent.main, 0.1),
+        border: `${theme.radii[1]}px solid ${theme.colors.accent.main}`,
```
what is the use case for radii then?

avatus replied:
It should be specific to `border-radius`. Also, I didn't see this before, but our theme has `theme.borders` so we could just do 
```suggestion
        border: `${theme.borders[2]} ${theme.colors.accent.main}`,
```

[check the sharedStyles here](https://github.com/gravitational/teleport/blob/master/web/packages/design/src/theme/themes/sharedStyles.ts#L53)

kimlisa commented:
friendly ping @rudream (i'll apply michaels change right before i merge)

kimlisa committed:
Final touches

- Added splitAwsIamArn helper that splits account id and its arn parts
- Require running a new script per change of each editable fields
  (includes roleArn)
- fix test to not emit "wrap with act" warning

kimlisa committed:
Merge branch 'master' into lisa/web-s3-bucket-warn

codingllama commented:
FYI, this broke `make lint-license`, which breaks Go PRs like #40337. It's likely that the linter isn't always running when it should.
